src/eppo-client.ts

```diff
@@ -50,7 +54,21 @@ export default class EppoClient implements IEppoClient {
     }
     const { variations, subjectShards } = experimentConfig;
     const shard = getShard(`assignment-${subjectKey}-${experimentKey}`, subjectShards);
-    return variations.find((variation) => isShardInRange(shard, variation.shardRange)).name;
+    const assignedVariation = variations.find((variation) =>
+      isShardInRange(shard, variation.shardRange),
+    ).name;
+    try {
+      this.assignmentLogger?.logAssignment({
+        experiment: experimentKey,
+        variation: assignedVariation,
+        timestamp: new Date().toISOString(),
+        subject: subjectKey,
+        ...subjectAttributes,
```

Comment:
Same comment here (from the python SDK review) re: having a separate property for subject attributes.